.github/workflows/publish.yml

```diff
@@ -0,0 +1,44 @@
+name: Publish to pub.dev
+
+## Caller of this workflow should use it as follows:
+## jobs:
+##   publish:
+##     uses: dart-lang/setup-dart/.github/workflows/publish.yml
+##     # with:
+##     #   working-directory: path/to/sub/directory
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: if specified, the workflow is required to be run in this environment (with additional approvals)
+        required: false
+        type: string
+      working-directory:
+        description: directory with-in the repository where the package is located (if not in the repository root)
+        required: false
+        type: string
+
+jobs:
+  publish:
+    name: 'Publish to pub.dev'
+    environment: ${{ inputs.environment }}
+    permissions:
+      id-token: write # This is required for requesting the JWT
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout repository
+      - uses: actions/checkout@v3
+      # Setup Dart SDK with JWT token
+      - uses: dart-lang/setup-dart@main
```

Comment:
Maybe `@main` is bad, it's not forward compatible.

We should just make it use a version tag..

Reply:
@mit-mit can we write `@1.4` here, merge this and create a `1.4` release/tag?

I think you're the only who has created releases of this github action, so maybe you're the owner? :rofl:

Reply:
Here we could use a commit hash too.
Updated with version for now, and the commented-out example above.

Reply:
Thinking about it, a commit hash here seems to be better for now. Updated it with the above comment too.